enclave/ecalls.cpp

```diff
@@ -39,7 +39,22 @@ int enclave_XGDMatrixCreateFromFile(const char *fname, int silent, DMatrixHandle
 int enclave_XGDMatrixCreateFromEncryptedFile(const char *fnames[], bst_ulong num_files, int silent, DMatrixHandle *out, char* usernames[]) {
   LOG(DEBUG) << "Ecall: XGDMatrixCreateFromEncryptedFile";
   // FIXME: check fnames, usernames ptrs
-  return XGDMatrixCreateFromEncryptedFile(fnames, num_files, silent, out, usernames);
+  char* filenames[num_files];
+  char* usrnames[num_files];
+  for (int i = 0; i < num_files; i++) {
+      const char* fname = fnames[i];
+      size_t nlen = strlen(fname) + 1;
```

Comment:
This `strlen` computation is unsafe. See #43 for how to handle this better.
Essentially, instead of computing the length inside the enclave, we should (i) pass in the lengths to the enclave (perhaps in a separate array), (ii) verify the buffer using this length, and (iii) copy the required number of characters into the enclave.

Reply:
Fixed!